docs/vocabulary.md

```diff
@@ -388,6 +389,75 @@ pipeline in corresponding documentation.
 | **Zuul**           | N/A       | N/A        | Job      | Pipeline | Trigger | Node (?)               |
 
+### Stages
+
+The list below is an attempt to create a mapping of common names for Stages, where a Stage is defined here as the unit of work one degree smaller than a Pipeline. (The effort of proposing a common vocabulary across CI/CD Tools and Technologies is not yet performed, but we'll use the most common term from the table above as the working term.)
+
+While CI/CD Tools and Technologies generally give developers broad leeway in naming and implementing their Stages, agreeing upon some common terms will help develop more pluggable Pipelines.
+
+The concepts below can be re-ordered in Pipeline implementations. For example, it is often the case that software linting (logically a test) is done prior to the Build Stage. Also, deployment to a staging environment is often done prior to the Release Stage, while the deployment for production often happens after the Release Stage.
+
+In the lists and table below, Software Artifacts includes: Documentation Source Files, Source Code, Baseline/Composition/Dependency Information, Infrastructure as Code.
+
+"Software Source" refers to human-readable source files that are inputs to the pipeline, such as: Source Code, Configuration Files, Documentation Source Files, Declared Dependencies, Baseline/Composition Information (i.e. lock files in source control)
+
+"Binary Source" refers to executables that are inputs to the pipeline, such as: Executable Software Dependencies, Container Images, Virtual Machine Images
+
+"Generated Software" refers to human-readable files that are outputs of the pipeline, such as: Compiled Software, Generated Configuration Files, Compiled Documentation, Baseline/Composition Information generated by the build. 
+
+"Generated Binaries" refers to executables that are outputs of the pipeline, such as: Executable Software, Container Images, Virtual Machine Images
+
+#### Build Stage
+* Semantics: Set up the build workspace and provision build resources. Download, retrieve, assemble, and/or compile software into an executable and testable format. Download, retrieve, assemble, and/or compile documentation into a consumable format.
+* Aliases: Compile
+* Inputs: Software Source, Binary Source
+* Outputs: Generated Software, Generated Binaries
+* Other Results and Side Effects: The build may also capture point-in-time details about the build environment and tools (compiler versions, package manager versions, task container versions in tekton, OS versions etc.) for anyone who wants to rebuild or understand a build.
+
+#### Test Stage
+* Semantics: Test, scan, verify, and lint software and documentation.
+* Aliases: Verify
+* Inputs: Software Source, Binary Source, Generated Software, Generated Binaries
+* Outputs: Test/Lint/Scan Records, Test/Lint/Scan Reports, Test/Lint/Scan Coverage Reports
+* Other Results and Side Effects: N/A
+
+#### Release Stage
```

Comment:
In CDEvents we have two events for this stage see https://github.com/cdevents/spec/blob/main/continuous-integration-pipeline-events.md.

If you have a situation where you want to test the packaged artifact before publishing. Would you then have stages as follows: `release` , `test`, `release`?

Reply:
I think that conceptually, releasing software happens after testing.  And publishing happens during the release phase/stage.

But then in "real live" pipelines, I've seen people put steps in all sorts of interesting sequences!  This is definitely an idealized set of stages that doesn't always map 100% to implementation.

Reply:
I guess I triggered on `packaged` being part of the release stage. I thought it was more common to do testing with the packaged artifact (to make sure nothing bad happened during package), but maybe that's only me.

Reply:
Go ahead and resolve this comment if you feel were digging to deep into details...

I presumed that the semantics parts was linked to the step PR like [package](https://github.com/cdfoundation/sig-interoperability/blob/8efa023bfcac366a2c56aad3c9787bf3acad81ec/docs/vocabulary.md#package), but maybe I was thinking to far here.